### PR TITLE
Bug fixed

### PR DESCRIPTION
**Pull request by shanushka**
bug fix-Edited Content Saved

Merge #1 before merging this

### CODE REVIEW

File: src/components/InpuField.css [hunk 1]
```diff
@@ -1,21 +1,50 @@
-.inputfield {
+.Add,
```
Comment (sanjeevkpandit) [inline src/components/InpuField.css]: class names should be of lowercase

```suggestion
.add
```

File: src/App.js [hunk 2]
```diff
@@ -1,18 +1,21 @@
...
 
-import Input from "./components/InputField";
-import Tab from "./components/TabBar";
-import TaskList from "./components/TaskList";
-import "./App.css";
-import { networkInterfaces } from "os";
 class App extends Component {
   constructor() {
     super();
+    const storageData = window.localStorage.getItem('todoData');
```
Comment (sanjeevkpandit) [inline src/App.js]: a new line after `super()`

File: src/components/TabBar.js [hunk 3]
```diff
@@ -1,15 +1,30 @@
...
-import "./TabBar.css";
+import React from 'react';
+
 const TabBar = props => {
   return (
     <div className='Tabs'>
-      <div className='tablinks' onClick={() => props.activeStage("Home")}>
+      <div
+        className={
+          props.activeState === 'Home' ? 'tablinks active' : 'tablinks'
```
Comment (sanjeevkpandit) [inline src/components/TabBar.js]: put the value in a separate variable and use it. It will make the code look cleaner.

Comment (sanjeevkpandit) [reply src/components/TabBar.js]: you can also have a look at the https://www.npmjs.com/package/classnames for defining dynamic or conditional class names.

File: src/components/Task.js [hunk 4]
```diff
@@ -15,12 +17,12 @@ class Task extends React.Component {
     if (!this.state.edit) {
       return (
         <div className='task clearfix'>
-          <span className={this.props.todo.done ? "tasks complete" : "tasks"}>
+          <span className={this.props.todo.done ? 'tasks complete' : 'tasks'}>
```
Comment (sanjeevkpandit) [inline src/components/Task.js]: same here

File: src/components/Task.js [hunk 5]
```diff
@@ -1,12 +1,14 @@
...
+import Input from './InputField';
 
 class Task extends React.Component {
-  constructor() {
+  constructor(props) {
     super();
+
     this.state = {
+      editedContent: props.todo.value,
       edit: false
```
Comment (sanjeevkpandit) [inline src/components/Task.js:21]: have an action name like `shouldEdit` or `isEditable` or something similar

File: src/components/Task.js [hunk 6]
```diff
@@ -34,23 +36,33 @@ class Task extends React.Component {
...
-              this.props.index,
-              this.handelEditButtonClick
-            );
-          }}
-          handleChange={this.props.changeInput}
-        />
+        <div className='task clearfix'>
+          <Input
+            userInput={this.state.editedContent}
+            value='Save'
```
Comment (sanjeevkpandit) [inline src/components/Task.js]:
```suggestion
            label='Save'
```

File: src/components/Task.js [hunk 7]
```diff
@@ -34,23 +36,33 @@ class Task extends React.Component {
...
+            }}
+            handleChange={evt => {
+              this.changeEditedContent(evt);
+            }}
+          />
+        </div>
       );
     }
   }
+  changeEditedContent = evt => {
```
Comment (sanjeevkpandit) [inline src/components/Task.js]: new line before function name

File: src/components/TaskList.js [hunk 8]
```diff
@@ -1,38 +1,45 @@
-import React from "react";
+import React from 'react';
 
-import Task from "./Task";
-import { isTag } from "postcss-selector-parser";
+import Task from './Task';
 
 const TaskList = props => {
-  console.log("ii", props.activeState);
+  console.log(props.todos);
```
Comment (sanjeevkpandit) [inline src/components/TaskList.js]: remove `console.log`

File: src/components/TaskList.js [hunk 9]
```diff
@@ -1,38 +1,45 @@
...
+
+  let filteredContent = props.todos.filter(todo => {
```
Comment (sanjeevkpandit) [inline src/components/TaskList.js]: use `const` if the value is not changing.
```suggestion
  const filteredContent = props.todos.filter(todo => {
```

File: src/components/TaskList.js [hunk 10]
```diff
@@ -1,38 +1,45 @@
...
-  } else if (props.activeState == "Incomplete") {
-    list = props.todos.filter(todo => !todo.done);
+  if (props.activeState == 'Completed') {
+    list = filteredContent.filter(todo => todo.done);
+  } else if (props.activeState == 'Incomplete') {
+    list = filteredContent.filter(todo => !todo.done);
+  } else {
+    list = [...filteredContent];
+  }
+  if (list.length != 0) {
```
Comment (sanjeevkpandit) [inline src/components/TaskList.js]: new line before `if`
```suggestion
  
  if (list.length != 0) {
```

File: src/components/TaskList.js [hunk 11]
```diff
@@ -1,38 +1,45 @@
...
+    return (
+      <div>
+        {list.map(todo => {
+          return (
+            <Task
+              key={todo.id}
+              index={todo.id}
```
Comment (sanjeevkpandit) [inline src/components/TaskList.js]: `id` should be better than `index`
```suggestion
              id={todo.id}
```

File: src/components/TaskList.js [hunk 12]
```diff
@@ -1,38 +1,45 @@
...
+  } else {
+    list = [...filteredContent];
+  }
+  if (list.length != 0) {
+    return (
+      <div>
+        {list.map(todo => {
+          return (
+            <Task
+              key={todo.id}
```
Comment (sanjeevkpandit) [inline src/components/TaskList.js:34]: `key` should be distinct. Use text message.
```suggestion
              key=`todo-&{todo.id}`
```

File: src/components/TaskList.js (hunk 11 above)
Comment (sanjeevkpandit) [reply src/components/TaskList.js]: You may not need this if you are passing `todo` as props.

File: src/components/TaskList.js [hunk 13]
```diff
@@ -1,38 +1,45 @@
...
+              changeInput={props.changeInput}
+              userInput={props.userInput}
+            />
+          );
+        })}
+      </div>
+    );
   } else {
-    list = [...props.todos];
+    return <div className="notodo">No Todo Lists To Show</div>;
```
Comment (sanjeevkpandit) [inline src/components/TaskList.js]: To make this more readable, return the single line statement in `if` and avoid using else.

```
if (list.length === 0) {
  return <div className="notodo">No Todo Lists To Show</div>;
}

return (
      <div>
        {list.map(todo => {
  ...
```

File: src/components/Task.js (hunk 5 above)
Comment (shanushka) [reply src/components/Task.js:21]: dai yo tara state haina

File: src/components/TaskList.js (hunk 12 above)
Comment (shanushka) [reply src/components/TaskList.js:34]: yesma k garnu parne ho

File: src/App.js [hunk 14]
```diff
@@ -1,86 +1,155 @@
...
-      userInput: "",
-      todos: []
+      searchField: '',
+      active: 'Home',
+      userInput: '',
+      todos: todos
     };
   }
-
+  /*
```
Comment (sanjeevkpandit) [inline src/App.js]: Improper docblock. Should be something like this:
```
/**
 * Description.....
 *
 * @params ...
 * @returns ...
 */
```

File: src/App.js [hunk 15]
```diff
@@ -1,86 +1,155 @@
...
-import { networkInterfaces } from "os";
+import './assets/css/layout.css';
+import Tab from './components/TabBar';
+import Input from './components/InputField';
+import TaskList from './components/TaskList';
+
+/** main component */
 class App extends Component {
+
+  /** main component */
```
Comment (prabhtz) [inline src/App.js]: Use format like:
```
/**
 * Function definition
 *
 * @param {type} paramName
 * @returns {returnType}
 */
```

File: src/App.js [hunk 16]
```diff
@@ -24,63 +47,96 @@ class App extends Component {
...
     });
+
     this.setState({ todos });
   };
 
-  handleComplete = index => {
+  handleComplete = id => {
     const todos = this.state.todos;
-    var indexx = todos.findIndex(todo => todo.id === index);
+    const indexx = todos.findIndex(todo => todo.id === id);
```
Comment (sanjeevkpandit) [inline src/App.js]:
```suggestion
    const index = todos.findIndex(todo => todo.id === id);
```

File: src/App.js [hunk 17]
```diff
@@ -24,63 +47,96 @@ class App extends Component {
...
+  handleSearch = evt => {
+    evt.preventDefault();
+
+    this.setState({
+      searchField: ''
+    });
+  };
+
   render() {
+
```
Comment (sanjeevkpandit) [inline src/App.js]: no newline required. you can remove this one.

File: src/App.js [hunk 18]
```diff
@@ -93,6 +149,7 @@ class App extends Component {
       </div>
     );
   }
+
```
Comment (sanjeevkpandit) [inline src/App.js]: no extra line required. you can remove this one.

File: src/assets/css/layout.css [hunk 19]
```diff
@@ -0,0 +1,215 @@
...
+.edit {
+  background: url(../images/edit.png) no-repeat;
+  background-size: cover;
+  height: 40px;
+  width: 40px;
+  margin: 0 4px;
+  border-radius: 50%;
+  border: none;
+}
+.add,
```
Comment (sanjeevkpandit) [inline src/assets/css/layout.css:40]: new line before every classname

File: src/components/InputField.js [hunk 20]
```diff
@@ -1,21 +1,26 @@
...
-    );
-  }
-}
+/**
+ *
+ *
+ * @param {objects} props
+ * @returns {Dom of Input field}
+ */
+
```
Comment (sanjeevkpandit) [inline src/components/InputField.js]: extra newline. you can remove this one.

File: src/components/InputField.js [hunk 21]
```diff
@@ -1,21 +1,26 @@
...
-            onChange={event => this.props.handleChange(event)}
-          />
-          <button>{this.props.value}</button>
-        </form>
-      </div>
-    );
-  }
-}
+/**
+ *
```
Comment (sanjeevkpandit) [inline src/components/InputField.js]: class name or some description of the component.

File: src/components/TabBar.js [hunk 22]
```diff
@@ -1,18 +1,38 @@
...
+  });
+
+  var completeClass = classNames('tablinks', {
+    active: props.activeState === 'Completed'
+  });
+
+  var inCompleteClass = classNames('tablinks', {
+    active: props.activeState === 'Incomplete'
+  });
   return (
```
Comment (sanjeevkpandit) [inline src/components/TabBar.js:18]: newline before return.

File: src/components/Task.js [hunk 23]
```diff
@@ -1,63 +1,97 @@
...
-import Input from "./InputField";
-import "./Task.css";
+import React from 'react';
 
+import Button from './Button';
+import Input from './InputField';
+import classNames from 'classnames';
+
+/**
+  * This is the main Class.
```
Comment (sanjeevkpandit) [inline src/components/Task.js:8]: format docblock

File: src/components/Task.js [hunk 24]
```diff
@@ -1,63 +1,97 @@
...
   render() {
+    const spanClass =classNames('tasks', {
+      completed:this.props.todo.done
+    });
+
+    var buttonClass = classNames({
+      'undo':this.props.todo.done,
+      'complete':!this.props.todo.done
+    })
     if (!this.state.edit) {
```
Comment (sanjeevkpandit) [inline src/components/Task.js:34]: new line before `if`

File: src/components/Task.js [hunk 25]
```diff
@@ -1,63 +1,97 @@
...
-          <span className={this.props.todo.done ? "tasks complete" : "tasks"}>
+        <li className='task clearfix'>
+          <span className={spanClass}>
             {this.props.todo.value}
           </span>
           <div class='buttons'>
             <button
-              className={this.props.todo.done ? "Undo" : "Complete"}
-              onClick={() => this.props.handleComplete(this.props.index)}
+              className={buttonClass }
```
Comment (sanjeevkpandit) [inline src/components/Task.js]: remove unnecessary extra spaces.

```suggestion
              className={buttonClass}
```

File: src/components/Task.js [hunk 26]
```diff
@@ -1,63 +1,97 @@
...
+            }}
+            handleChange={this.changeEditedContent}
+          />
+      </li>
       );
     }
   }
 
+  /**
+   *
```
Comment (sanjeevkpandit) [inline src/components/Task.js]: Put some description or remove unnecessary extra lines

File: src/components/TaskList.js [hunk 27]
```diff
@@ -1,38 +1,55 @@
-import React from "react";
+import React from 'react';
 
-import Task from "./Task";
-import { isTag } from "postcss-selector-parser";
+import Task from './Task';
 
+/**
+ *
```
Comment (sanjeevkpandit) [inline src/components/TaskList.js]: description required

File: src/components/TaskList.js [hunk 28]
```diff
@@ -1,38 +1,55 @@
...
+ *
+ * @param {*} props
```
Comment (sanjeevkpandit) [inline src/components/TaskList.js:8]: somewhere it's `{objects}` and somewhere `{*}`. have a consistent value.

File: src/components/TaskList.js [hunk 29]
```diff
@@ -1,38 +1,55 @@
...
+ * @returns
+ */
 const TaskList = props => {
-  console.log("ii", props.activeState);
+
```
Comment (sanjeevkpandit) [inline src/components/TaskList.js:12]: remove extra line

**Review by sanjeevkpandit — APPROVED** [review]

**Pull request merged**